Added Proper Documentation and Readme Files

prasad-chavan1 opened the pull request.

rohan472000 commented:
While raising PR, give full description, what you changed. It will be better for me to understand quickly.

prasad-chavan1 commented:
> While raising PR, give full description, what you changed. It will be better for me to understand quickly.

## Pull Request Description

**Title:**  Added Documentation and Linear Search Algorithm.

### Description
i've added README files to all directories of main branch and added linear search algorithm.

### Changes Made
- Added Docs
- Added Linear Search .Java

### Checklist:
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] All new README files are placed inside an appropriate directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Java naming conventions.
* [x] All function parameters and return values are annotated with Java.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.

rohan472000 commented:
Add these description above.
